partner_communication_switzerland/models/mail_tracking_event.py

```diff
@@ -100,21 +100,53 @@ def process_reject(self, tracking_email, metadata):
     def _invalid_email(self, tracking_email):
         """
         Sends invalid e-mail communication.
+        In particular cases avoid sending the communciation because it will append later on
+        in an other process :
+        - if the invalid event came from an email linked to a correspondence (b2s)
+        - if the invalid event came from an email link to the onboarding confirmation
         """
+        
+        # invalid config to send to the partner
         invalid_comm = self.env.ref("partner_communication_switzerland.wrong_email")
+
+        # correspondence mail should return b2s email not read
+        correspondence_email = tracking_email.filtered(
+            lambda x: x.mail_id.model == "partner.communication.job" and
+                      self.env["partner.communication.job"].browse(
+                          x.mail_id.res_id).model == "correspondence"
+        )
+
+        # other special case for onboarding welcome not read
+        onboarding_welcome_config = self.env.ref(
+            "partner_communication_switzerland.config_onboarding_sponsorship_confirmation")
+
+        onboarding_welcome_email = tracking_email.filtered(
+            lambda x: x.mail_id.model == "partner.communication.job" and
+                      self.env["partner.communication.job"].browse(
+                          x.mail_id.res_id).config_id == onboarding_welcome_config
+        )
+
+        # even if the specific communication is not yet going out to the partner with
+        # create a not on his/her odoo profile page
+        tracking_email.partner_id.message_post(
+            body=_("The email couldn't be sent due to invalid address"),
+            subject=tracking_email.name
+        )
+
+        # in those case specific communication will be sent in another process
+        # return to avoid duplication
+        if correspondence_email or onboarding_welcome_email:
+            return
+
         partner_id = tracking_email.partner_id.id
         if partner_id:
             self.env["partner.communication.job"].create(
                 {
-                    "config_id": invalid_comm.id,
+                    "config_id": invalid_comm,
```

Comment:
The previous version was correct. `invalid_comm` is a record and in the values dictionary we want to put the `id` of the record.

Reply:
Sorry for that... thank you